helper/helper_test.go

```diff
@@ -98,4 +294,234 @@ var _ = Describe("helper tests", func() {
+			}
+
+			helper := newHelperTest(oidBatch, expectedScenario)
+			err := doRestoreAgentInternal(helper)
+			Expect(err).To(BeNil())
+		})
+		It("skips batches if skip file is discovered with single datafile", func() {
+			*singleDataFile = true
+			*isResizeRestore = false
+			*tocFile = testTocFile
+
+			// Although pure concept would be to mock TOC file as well, to keep things simpler
+			// let's use real TOC file here
+			writeTestTOC(testTocFile)
+			defer func() {
+				_ = os.Remove(*tocFile)
+			}()
+
+			oidBatch := []oidWithBatch{
+				{100, 0},
+				{200, 0},
+				{200, 1},
+				{200, 2},
+			}
+
+			expectedScenario := []helperTestStep{
+				{"mock_100_0", true, -1, false, "Can open pipe for table 100, check_skip_file shall not be called"},
+				{"mock_200_0", true, -1, false, "Can open pipe for table 200, check_skip_file shall not be called"},
+				{"mock_200_1", false, 200, true, "Can not open pipe for table 200, check_skip_file shall called, skip file exists"},
+				{"mock_200_2", true, -1, false, "Went to the next batch, Can open pipe for table 200, check_skip_file shall not be called"},
+			}
+
+			helper := newHelperTest(oidBatch, expectedScenario)
+			err := doRestoreAgentInternal(helper)
+			Expect(err).To(BeNil())
+		})
+		It("calls Wait in waitForPlugin doRestoreAgent for single data file", func() {
+			*singleDataFile = true
+			*isResizeRestore = false
+			*tocFile = testTocFile
+
+			// Although pure concept would be to mock TOC file as well, to keep things simpler
+			// let's use real TOC file here
+			writeTestTOC(testTocFile)
+			defer func() {
+				_ = os.Remove(*tocFile)
+			}()
+
+			oidBatch := []oidWithBatch{{100, 0}}
+			expectedScenario := []helperTestStep{
+				{"mock_100_0", true, -1, false, "Some pipe shall be created, out of interest for this test although"},
+			}
+			helper := newHelperTest(oidBatch, expectedScenario)
+
+			err := doRestoreAgentInternal(helper)
+			Expect(err).ToNot(HaveOccurred())
+
+			// Check that plugin command's Wait was acually called and only once
+			Expect(helper.restoreData.waitCount).To(Equal(1))
+		})
+		It("calls waitForPlugin doRestoreAgent for resize and no single data file ", func() {
+			Expect(*singleDataFile).To(Equal(false))
+
+			oidBatch := []oidWithBatch{{100, 0}}
+			expectedScenario := []helperTestStep{
+				{"mock_100_0", true, -1, false, "Some pipe shall be created, out of interest for this test although"},
+			}
+
+			helper := newHelperTest(oidBatch, expectedScenario)
+
+			err := doRestoreAgentInternal(helper)
+			Expect(err).ToNot(HaveOccurred())
+
+			// Check that plugin command's Wait was acually called and only once
+			Expect(helper.restoreData.waitCount).To(Equal(1))
+		})
+		It("calls waitForPlugin doRestoreAgent for reduce cluster and no single data file ", func() {
+			Expect(*singleDataFile).To(Equal(false))
+			*destSize, *origSize = *origSize, *destSize
```

Comment:
What is difference from test "skips batches if skip file is discovered with resize restore" above?

Reply:
Sorry, don't understand the question. The scenario is totally different.

Reply:
I didn't ask about the whole scenario, but only about working with the `destSize` and `origSize`.

Reply:
Then I don't understand your question.

Reply:
test "skips batches if skip file is discovered with resize restore" explicitly sets `*origSize = 3` and `*destSize = 5`, while this test does same by another way

Reply:
Similar to previous comment, the number of batches is different and skip file exists in one of them and one batch with no skip file in the other.

Reply:
I repeat it again more slowly:
- test "skips batches if skip file is discovered with resize restore" explicitly sets `*origSize = 3` and `*destSize = 5`
- before test "calls waitForPlugin doRestoreAgent for reduce cluster and no single data file " BeforeEach finction explicitly sets `*origSize = 5` and `*destSize = 3` and in test these are swapped, resulting `*origSize = 3` and `*destSize = 5` as above

Reply:
Okay. What is the question or suggestion?

Reply:
Make it the same in both tests, for example, just explicitly set the values `*origSize = 3` and `*destSize = 5`.